services/dashboard/routes/overview.go

```diff
@@ -16,132 +14,172 @@ import (
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/chi/v5/middleware"
 	"github.com/gorilla/websocket"
+	"github.com/samber/lo"
 
-	"sensorbucket.nl/sensorbucket/internal/pagination"
 	"sensorbucket.nl/sensorbucket/internal/web"
-	"sensorbucket.nl/sensorbucket/services/core/devices"
-	"sensorbucket.nl/sensorbucket/services/core/measurements"
-	coretransport "sensorbucket.nl/sensorbucket/services/core/transport"
+	"sensorbucket.nl/sensorbucket/pkg/api"
 	"sensorbucket.nl/sensorbucket/services/dashboard/views"
 )
 
 type middlewareFunc = func(next http.Handler) http.Handler
 
-func CreateOverviewPageHandler() http.Handler {
-	r := chi.NewRouter()
+type OverviewRoute struct {
+	router chi.Router
+	client *api.APIClient
+}
+
+func CreateOverviewPageHandler(client *api.APIClient) *OverviewRoute {
+	t := &OverviewRoute{
+		client: client,
+		router: chi.NewRouter(),
+	}
+	t.SetupRoutes(t.router)
+	return t
+}
+
+func (t OverviewRoute) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
are the basic auth credentials inserted somewhere else?

Reply:
Good one! Not yet. Must add!